### DISCUSSION TITLE
how to import js functions and classes to the main compiled them.js file?

### DISCUSSION THREAD

**Discussion by yoscbd** [holger1411/understrap#1825] [Q&A] [answered]
Hi :)

I know i can use "\understrap\src\js\custom-javascript.js" to add my own js code that will be merged to the main **"understrap-scripts"** in "inc/enqueue.php"  and lather have this code included in **theme.min.js** file

How can I include another file other then **"custom-javascript.js"** that will be merged and complied to  **theme.min.js** in the same way as "custom-javascript.js"?

**my goal is to have this file compiled to the "thems.min.js" so I can avoid another request to the server for my js file,
This is why i don't want to use wp_enqueue_script () as it will create another request to the server.

I want to be able to include a custom js file containing js classes and methods,
This is the content of my custom js file i want to include:

```
import $ from "jquery"

class MyNotes {
  constructor() {
    this.events()
  }

  events() {
    $(".delete-note").on("click", this.deleteNote)
  }

  // Methods will go here
  deleteNote() {
    alert("you clicked delete")
  }
}

export default MyNotes
```

Most of the time when building other NPM dev env (not Understrap) I use a main "index.js" file that import other files and methods that are compiled to a single js file.

so in my main index.js file i import any other js files or classes i want to use:

```
// Our modules / classes
import Search from "./modules/Search"
import Search from "./modules/Banner"
import Search from "./modules/REST"
```

what will be the best practice to achieve my goal?

Thank you!

**Comment by IanDelMar** [comment, marked as answer]
You can add other js files by adding them to the rollup.config.js.

I assume this is your 'other-custom-javascript.js':
```js
import $ from "jquery"

class MyNotes {
  constructor() {
    this.events()
  }

  events() {
    $(".delete-note").on("click", this.deleteNote)
  }

  // Methods will go here
  deleteNote() {
    alert("you clicked delete")
  }
}

export default MyNotes
```
and this is your  'custom-javascript.js':
```js
import MyNotes from "./other-custom-javascript";

const notes = new MyNotes();
```
then your 'rollup.config.js' would look like so:
```js
module.exports = {
	input: [
		path.resolve( __dirname, `../js/bootstrap.js` ),
		path.resolve( __dirname, '../js/skip-link-focus-fix.js' ),
		path.resolve( __dirname, '../js/other-custom-javascript.js' ),
		path.resolve( __dirname, '../js/custom-javascript.js' ),
	],
	...
};
```

**Comment by yoscbd** [comment]
Thank you!!!,
 that was super helpful and solved my issue.

I have edited:
understrap\src\build\rollup.config.js

and followed the steps you have mentioned and now all is working as expected.
 :)
